src/segments/git.go

```diff
@@ -706,14 +709,23 @@ func (g *Git) WorktreeCount() int {
 func (g *Git) getRemoteURL() string {
 	upstream := regex.ReplaceAllString("/.*", g.Upstream, "")
 	if len(upstream) == 0 {
-		upstream = "origin"
+		upstream = UPSTREAM
+	}
+	insteadOfs := g.props.GetStringArray(InsteadOfURLS, []string{})
+	if len(insteadOfs) == 1 && insteadOfs[0] == "*" {
```

Comment:
I read the PR and the issue linked to it. Since this was about performance, I was not sure if having an option like `remote_source` would suffice.

Although `remote_source` might give the user the freedom to either take the performance hit or, if the icon logic is implemented with the `RawUpstreamURL` instead of using the cleanup one, the user can add the custom icon themselves with the `upstream_icons` property.